### PR TITLE
call user defined drop impls

### PR DESCRIPTION
**Pull request by oli-obk**

### CODE REVIEW

File: src/interpreter/terminator/mod.rs [hunk 1]
```diff
@@ -478,29 +508,36 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
...
             return Ok(());
         }
         trace!("-need to drop {:?} at {:?}", ty, lval);
 
-        // TODO(solson): Call user-defined Drop::drop impls.
-
         // special case `Box` to deallocate the inner allocation
-        // FIXME: if user defined Drop impls work, then this can go away, since the stdlib calls
-        // heap::deallocate
         if let ty::TyBox(contents_ty) = ty.sty {
```
Comment (solson) [inline src/interpreter/terminator/mod.rs]: Merge this `if let` into the `match` below.

**Comment by solson** [comment]
We should be able to fix this now, right? https://github.com/solson/miri/blob/master/tests/run-pass/vecs.rs#L1

**Comment by solson** [comment]
In fact you can probably just `git revert` this whole commit https://github.com/solson/miri/commit/366c793306609f4a80e7977be766cbc7e9c2b3be.

**Comment by oli-obk** [comment]
indeed... reverting that commit simply worked...

**Pull request merged**